hudi-utilities/src/main/java/org/apache/hudi/utilities/sources/InputBatch.java

```diff
@@ -48,6 +49,9 @@ public String getCheckpointForNextBatch() {
   }
 
   public SchemaProvider getSchemaProvider() {
+    if (schemaProvider == null) {
+      throw new HoodieException("Please provide a valid schema provider class!");
+    }
```

Comment:
This change affects the TestHoodieDeltaStreamer#testBulkInsertsAndUpsertsWithSQLBasedTransformerFor2StepPipeline, and the UT should be changed as well. You could click the details below to get travis detail message.

Reply:
Working on it.

Reply:
@leesf Done with the changes, please review.